stable/mongodb-replicaset/templates/_helpers.tpl

```diff
@@ -81,9 +81,13 @@ Create the name for the key secret.
 Allow the release namespace to be overridden for multi-namespace deployments in combined charts.
 */}}
 {{- define "mongodb-replicaset.namespace" -}}
-  {{- if and ((hasKey .Values "global") .Values.global.namespaceOverride) -}}
-    {{- .Values.global.namespaceOverride -}}
+  {{- if .Values.global -}}
```

Comment:
This could be combined into a single if/else to reduce duplication with `{{- if and .Values.global .Values.global.namespaceOverride }}`

Reply:
Actually it looks like it will check both conditions, so even when global one is not valid, still it checks the second condition
```
executing ".namespace" at <.Values.global.namespaceOverride>: nil pointer evaluating interface {}.namespaceOverride
```

Reply:
now I think about it, it was the reason that I used `hasKey` in first place

Reply:
Yeah, there is no lazy if conditional

Reply:
Strange, never seen a language without a short-circuiting AND operator. Confirmed it in [docs](https://golang.org/pkg/text/template/#hdr-Functions).